collect_app/src/main/java/org/odk/collect/android/map/GoogleMapFragment.java

```diff
@@ -370,47 +416,97 @@ protected void updateFeatures() {
             if (parts.length >= 2) {
                 sd = Double.parseDouble(parts[1]);
             }
-        } catch (NumberFormatException e) { /* ignore */ }
+        } catch (NumberFormatException e) {
+            Timber.w("Assertion violated: Marker.getSnippet() did not contain two numbers");
```

Comment:
Nit: This isn't technically an assertion because it doesn't abort, maybe just say `"Error: ..."`?

Reply:
I'm happy to change this message.  A little context on how we got here, though:

When I originally wrote this exception clause, it ignored the exception because there should be no way for execution to reach this point; and if somehow it does, the best way to proceed is to use zero, which is what happens when the exception is ignored.

It was then suggested that this should log a warning instead of doing nothing.  In phrasing the warning message, I was looking for a way to express that this is a "This should never happen" situation—execution should never reach this point.  If it does, it is due to an internal inconsistency in the program; it cannot happen due to bad user input.  This is just the type of circumstance where one would normally `assert`.  But I assumed that we would never want the app to crash, so instead of using an actual assertion, I used the message to say that it was just like an assertion violation.

Reply:
@yanokwa What's ODK's conventional practice for assertion-like checks?

Reply:
There's no practice that I'm aware of. And for those situations, I tend to match whatever is in the file.

Reply:
Is there a way to log this as a bug report, to get a stack trace when it happens and diagnose?

Reply:
Paging @grzesiek2010!

Reply:
Okay.  @yanokwa What's your preference for whether a "this shouldn't happen" situation should crash the app or not?

Reply:
To be clear, I definitely don't think it should crash, since it seems like a local problem, I just think we should log a bug report so we can diagnose it.

Reply:
According to this method https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/application/Collect.java#L303
Log.WARN and Log.ERROR are logged.

Reply:
Great!